File: SIMPLEAPI_Demo/SIMPLEAPI_Demo/ConsultaEstadoEnvioDTE.cs
using System;
//using ChileSystems.DTE.Engine.Enum;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
//using static SIMPLE_API.Enum.Ambiente;
using Newtonsoft.Json;

namespace SIMPLEAPI_Demo
{
    public partial class ConsultaEstadoEnvioDTE : Form
    {
        Handler handler = new Handler();
        public ConsultaEstadoEnvioDTE()
        {
            InitializeComponent();
        }

        private void ConsultaEstadoAvanzadoDTE_Load(object sender, EventArgs e)
        {
            handler.configuracion.LeerArchivo();
            textRUTEmpresa.Text = handler.configuracion.Empresa.RutCuerpo.ToString();
            textDVEmpresa.Text = handler.configuracion.Empresa.DV;
        }

        private async void botonConsultar_Click(object sender, EventArgs e)
        {
            long trackId = long.Parse(textTrackID.Text);
            try
            {
                if (radioEnvioDTE.Checked)
                {
                    var responseEstadoDTE = await handler.ConsultarEstadoEnvioDTEAsync(radioProduccion.Checked ? SimpleAPI.Enum.Ambiente.AmbienteEnum.Produccion : SimpleAPI.Enum.Ambiente.AmbienteEnum.Certificacion, trackId);
                    textRespuesta.Text = responseEstadoDTE.ResponseXml;
                }
                else
                {
                    var responseEstadoDTE = await handler.ConsultarEstadoEnvioBoletaAsync(radioProduccion.Checked ? SimpleAPI.Enum.Ambiente.AmbienteEnum.Produccion : SimpleAPI.Enum.Ambiente.AmbienteEnum.Certificacion, trackId);
                    textRespuesta.Text = JsonConvert.SerializeObject(responseEstadoDTE, Formatting.Indented);
                }
            }
            catch (Exception ex)
            {
                MessageBox.Show("Ha ocurrido un error:" + ex);
            }
        }
    }
}

File: SIMPLEAPI_Demo/SIMPLEAPI_Demo_VB/ConsultaEstadoEnvioDTE.vb
Imports Newtonsoft.Json
Imports SIMPLE_API.Enum.Ambiente

Public Class ConsultaEstadoEnvioDTE

    Dim handler As Handler = New Handler()

    Private Sub ConsultaEstadoEnvioDTE_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        handler.configuracion.LeerArchivo()
        textRUTEmpresa.Text = handler.configuracion.Empresa.RutCuerpo.ToString()
        textDVEmpresa.Text = handler.configuracion.Empresa.DV
    End Sub

    Private Sub botonConsultar_Click(sender As Object, e As EventArgs) Handles botonConsultar.Click
        Dim trackId As Long = Long.Parse(textTrackID.Text)

        Try

            If radioEnvioDTE.Checked Then
                Dim responseEstadoDTE = handler.ConsultarEstadoEnvio(If(radioProduccion.Checked, AmbienteEnum.Produccion, AmbienteEnum.Certificacion), trackId)
                textRespuesta.Text = responseEstadoDTE.ResponseXml
            Else
                Dim responseEstadoDTE = handler.ConsultarEstadoEnvioBoleta(If(radioProduccion.Checked, AmbienteEnum.Produccion, AmbienteEnum.Certificacion), trackId)
                textRespuesta.Text = JsonConvert.SerializeObject(responseEstadoDTE, Formatting.Indented)
            End If

        Catch ex As Exception
            MessageBox.Show($"Ha ocurrido un error: {ex}")
        End Try
    End Sub
End Class